### ISSUE TITLE
git commit hash and NeptuneObserver

### ISSUE THREAD

**Issue by wjaskowski** [neptune-ml/neptune-contrib#55]
I have noticed that NeptuneObserver doesn't respect the git hash and the source file names that sacred retrieved but has its own methods to retrieve them. In general, this is wrong because it breaks the RunObserver contract from the user point of view. Thus surprised me a lot and I spent o lot of time to understand that.

Although, I could live with that but in certain situations, NeptuneObserver fails to get the proper git commit hash while sacred doesn't. I couldn't track down the exact reason for that but it happens when 
1) I run my experiment script (git repo) from a different script
2) my working directory (outside git repo) is different from my experiment script directory (git repo)

A similar situation was with sources. One solution was to set manually the base_dir in NeptuneObserver (but notice: sacred has its own base_dir mechanism - why not to use it?). But when I set my base_dir to e.g. `/home/name/x/y`, then the whole tree is then in the sources (starting with `home`), which is not ideal. 

A second (not very elegant) workaround is to use `os.chdir()` - then the sources are gathered correctly but this doesn't affect git hash which is still missing, unfortunately.

**Comment by pitercl** [comment]
Thanks for the writeup!

We'll get to it, but probably not sooner than a week from now.

@jakubczakon Will you take a look?

**Comment by wjaskowski** [comment]
Let me propose a solution that works for git hashes:
```python
    def __init__(self, project_name, api_token=None, source_extensions=None):
        neptune.init(project_qualified_name=project_name, api_token=api_token)
        self.resources = {}
        if source_extensions:
            self.source_extensions = source_extensions
        else:
            self.source_extensions = ['.py', '.R', '.cpp', '.yaml', '.yml']

    def started_event(self, ex_info, command, host_info, start_time, config, meta_info, _id):
        neptune.create_experiment(name=ex_info['name'],
                                  params=_flatten_dict(config),
                                  upload_source_files=_get_filepaths(dirpath=ex_info['base_dir'],
                                                                     extensions=self.source_extensions),
                                  properties={'mainfile': ex_info['mainfile'],
                                              'dependencies': str(ex_info['dependencies']),
                                              'sacred_id': str(_id),
                                              **_str_dict_values(host_info),
                                              **_str_dict_values(_flatten_dict(meta_info)),
                                              **_str_dict_values(_flatten_dict(ex_info))},
                                  git_info=neptune.utils.get_git_info(ex_info['base_dir'])
                                  )
```

Unfortunately, the sources still have this weird structure starting with `/home/` but I can live with it ;)

**Comment by jakubczakon** [comment]
That was indeed a very nice suggestion @wjaskowski.
I used it and fixed the `/home` issue by using the newly available functionality that accepts stuff like: `upload_source_files=[`**/*.py`]`. It was merged with master in #57

**Comment by wjaskowski** [comment]
Great. Thx!
